### PR TITLE
feat(overview): add PC specs display, update checker, and UI improvem…

### PR DESCRIPTION
**Pull request by evildarkarchon**
…ents

- Add IPcInfo/IPcInfoService to display OS, RAM, CPU, and GPU info on Overview
- Add IUpdateChecker/UpdateChecker with Nexus/GitHub update detection
- Implement update banner in MainWindow with version links
- Add SettingsViewModel with update channel and log level options
- Add ExternalLinks constants for Nexus, Discord, and GitHub URLs
- Redo About page with app info, logos, and social links
- Show mod manager detection details with warning tooltip
- Add clickable game path to open folder
- Fix CCC warning logic to only show when Address Library is present
- Add ShowMissingCccWarning and ShowMissingPluginsWarning flags
- Add registry GetQwordValue for DWORD registry values
- Add tests for CCC warning behavior and update/PC info services

---

> [!NOTE]
> **Medium Risk**
> Touches census warning semantics, registry/OS APIs for PC info, and outbound HTTP for updates; broad new tests reduce regression risk but the behavioral surface area for Overview and startup networking is non-trivial.
> 
> **Overview**
> Delivers **Milestone 2 shell work** aligned with the PR title: **async update detection** (`IUpdateChecker` / Nexus + GitHub) drives a **main-window InfoBar** from `MainViewModel`, with **Settings** persisting update channel and log level. **Overview** gains **PC specs** via `IPcInfo` / registry+P/Invoke, plus census/UI flags **`ShowMissingCccWarning`** / **`ShowMissingPluginsWarning`** (with tests clarifying CCC dialog behavior vs Address Library / unreadable `plugins.txt`). **About** is rebuilt around `ExternalLinks` and assets; **`GetQwordValue`** extends registry reads for GPU VRAM.
> 
> The diff also adds **large `.kilo/plans/`** write-ups for M2–M4 (scanner, downgrader, archive patcher) and **substantial `CMToolkit.Core.Tests`** coverage (downgrade, archive patch, F4SE analyzer, scanner index, census edge cases, `PeExportReader` bounds). **CodeRabbit path filters** skip more non-product paths; **VS Code / CHANGELOG** are mostly formatting.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 61337d15c7ea9cae603be96813d79b08d7602307. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

## Summary by CodeRabbit

* **Style**
  * Updated the solution’s project entries to explicitly declare x64 platform support and added standardized solution configuration listings (including Any CPU and x64).

* **Note**
  * This change is limited to solution configuration/formatting updates and has no expected user-visible or functional impact.

### CODE REVIEW

File: CMToolkit/ViewModels/OverviewViewModel.cs [hunk 1]
```diff
@@ -182,6 +281,21 @@ private void PopulateFrom(GameSession session, OverviewCensusResult result)
         }
 
         ProblemCount = result.Problems.Count;
+        if (isInitialLoad)
+        {
+            if (result.ShowMissingCccWarning)
+            {
+                _pendingInitialWarnings.Add(
+                    "Fallout4.ccc not found.\nCC files may not be detected. Verifying Steam files or reinstalling should fix this.");
+            }
```
[the remark this answers is not in the document]
Comment (evildarkarchon) [reply CMToolkit/ViewModels/OverviewViewModel.cs]: Address Library and CCC file are unrelated
